cmd/snap-seccomp/main.go

```diff
@@ -466,137 +457,6 @@ func readNumber(token string) (uint64, error) {
 	return strconv.ParseUint(token, 10, 64)
 }
 
-// Be very strict so usernames and groups specified in policy are widely
-// compatible. From NAME_REGEX in /etc/adduser.conf
-var userGroupNamePattern = regexp.MustCompile("^[a-z][-a-z0-9_]*$")
-
-func findUid(username string) (uint64, error) {
-	if !userGroupNamePattern.MatchString(username) {
-		return 0, fmt.Errorf("%q must be a valid username", username)
-	}
-	user, err := user.Lookup(username)
-	if err != nil {
-		return 0, err
-	}
-
-	return strconv.ParseUint(user.Uid, 10, 64)
-}
-
-// hrm, user.LookupGroup() doesn't exist yet:
-// https://github.com/golang/go/issues/2617
-//
-// Use implementation from upcoming releases:
-// https://golang.org/src/os/user/lookup_unix.go
-func lookupGroup(groupname string) (string, error) {
```

Comment:
why are we doing it this way instead of using go's LookupGroup? it's a lot simpler fwiw

Reply:
Because it is not available in golang 1.6

Reply:
I understand that, but we can copy the LookupGroup function from go 1.9 or whatever, as we've done before for similar things

Reply:
ah. I looked at lookup_unix, but this is cgo_lookup_unix. Heh. Sorry.

Reply:
This is exactly that. I just took the existing copy @jdstrand added to snap-confine and made it public.